woocommerce/google-listings-and-ads
# Update UI to handle Fetching Product Status with Action Scheduler Jobs

### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/google-listings-and-ads/issues/2146 & https://github.com/woocommerce/google-listings-and-ads/issues/2250

While we're refactoring the backend logic for the product stats and moving it to the Action Scheduler Job, we also need to update the UI to reflect these changes. This involves adjusting how the UI behaves when the job is scheduled or running and the product stats aren't available yet. Hence, we'll need to make a call to `wc/gla/mc/product-statistics/` every 30 seconds to check if the stats calculation has been completed.


https://github.com/woocommerce/google-listings-and-ads/assets/2488994/e4aff324-1074-4513-b645-f80b0deb18e9



In summary, this PR introduces the following:

- It will continuously call `wc/gla/mc/product-statistics/` every 30 seconds until the `update_merchant_product_statuses` job is finished and the stats can be retrieved.

Example response when loading:
```
{
    "timestamp": 1710711097,
    "statistics": null,
    "scheduled_sync": 0,
    "loading": true,
    "error": null
}
```
Example response when it is completed:

```
{
    "timestamp": 1710711097,
    "statistics": {
      "active" : 10
      "disapproved" : 1
      "expiring" : 2 
      "not_synced" : 3
      "pending" : 4
    },
    "scheduled_sync": 0,
    "loading": false,
    "error": null
}
```

2. Once the fetching process is completed, it will invalidate the Product Feed Query, ensuring that the Product MC Statuses are updated.

### Detailed test instructions:


1. Make a call to `GET wp-json/wc/gla/mc/product-statistics/refresh`.
2. Navigate to `wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed` or refresh the product feed page.
3. Monitor the network requests and observe that the network request `wc/gla/mc/product-statistics` is fetched every 30 seconds until the loading status becomes false.
4. Verify that the product feed request is issued.
5. Confirm that the Overview Stats are loaded.
6. Ensure that the product feed displays the updated MC statuses.


### Additional details:

I was doubting whether it would be better to use an AppSpinner for each stats value or simply use the `SummaryListPlaceholder`. In the end, I decided to go with the placeholder because it maintains a more consistent look with what we currently have. However, I'm open to hearing other opinions.

AppSpinner:

https://github.com/woocommerce/google-listings-and-ads/assets/2488994/5c5b50c1-0e80-49e9-9771-1f000aa98621

PlaceHolder:

![image](https://github.com/woocommerce/google-listings-and-ads/assets/2488994/36fd81a6-d8e5-4b2f-aa53-e47d57600816)



### Changelog entry

>
